Python/Product/VSInterpreters/PipPackageManager.cs

```diff
@@ -458,17 +459,18 @@ CancellationToken cancellationToken
                         try {
                             if ((await proc) == 0) {
                                 if (_pipListHasFormatOption) {
-                                    var json = string.Join(Environment.NewLine, proc.StandardOutputLines);
                                     try {
-                                        var data = JArray.Parse(json);
+                                        var data = JToken.ReadFrom(new JsonTextReader(new StringListReader(proc.StandardOutputLines)));
```

Comment:
I wonder if it would be better to change `ProcessOutput` to store the data as a single chunk in the first place (e.g. as `MemoryStream`? so that it can be fed directly to anything that expects a stream or a reader), and reimplement `StandardOutputLines` as lazy iterator over that.

Reply:
I think the storing it as the single chunk is the problem. A list of strings can be relocated by the GC, while a chunk approaching GB is going to be hard to allocate (I don't have sufficient info from the crash data, but my guess is that it's a failed large allocation rather than a leak, since it was in the `string.Join`).

If it's still a problem here, we can probably create an adapter between the `Redirector` and `JsonTextReader`, but since it's a push vs. pull model adapter it's going to be much more complicated.